task/mongo-optimization-task.js

```diff
@@ -240,22 +225,18 @@ async function task_3_1(db) {
         {
             "$lookup" : {
                 "from" : "clientCriteria",
-                "localField" : "criteria_value",
-                "foreignField" : "value",
+                let: { item: "$criteria_value" },
+                pipeline: [
+                    {
+                        "$match" : {
+                            "versions.initiativeId" : ObjectId("58af4da0b310d92314627290"),
+                            $expr: { $eq: [ "$value",  "$$item" ] }
+                        }
+                    }
```

Comment:
Additional projection, compound index and `limit` at the end can make the query faster.

Reply:
@IlyaMokin thank you! done - https://travis-ci.org/github/AlinaNH/db-assignments/builds/726557102